### PR TITLE
The world's best dad jokes - Annika Melin

### PR DESCRIPTION
**Pull request by ankimelin**
My reflections on how this week's project turned out:
I'm happy I built something on a simple API, as I quite quickly had a base application to continue on. I wanted to practise text input, and it was fun to implement a share function. Tried out "try/catch" and "finally" for the first time.

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,26 +1,151 @@
...
+      method: 'GET',
+      headers: {
+        "Accept": "application/json"
+      }
+    })
+      .then(response => response.json())
+      .then(json => setJoke(json.joke))
+  }
+
+  const searchJoke = () => {
```
Comment (EmOlsz) [inline App.js:87]: From naming point of view, `getJoke()` and `searchJoke()` functions are not very descriptive. I would rather suggest something like `getJoke()` and `getSearchedJoke()`

File: App.js [hunk 2]
```diff
@@ -1,26 +1,151 @@
...
+      })
+  }
+
+  const shareJoke = async () => {
+    try {
+      const result = await Share.share({
+        message:
+          `${joke}`,
+      })
+      result.action === Share.sharedAction ? (alert("Joke shared")) :
```
Comment (EmOlsz) [inline App.js:114]: No need to wrap `alert()` inside extra set of parentheses, this is completely enough :

```
result.action === Share.sharedAction ? alert("Joke shared") : 
```

File: App.js [hunk 3]
```diff
@@ -1,26 +1,151 @@
...
+      .catch(() => {
+        alert("No such joke")
+      })
+      .finally(() => {
+        setSearch("")
+        setSearch()
+      })
+  }
+
+  const shareJoke = async () => {
```
Comment (EmOlsz) [inline App.js:108]: Great job with `async await`:)
For further explanation of this topic, check out https://medium.com/javascript-in-plain-english/async-await-javascript-5038668ec6eb

File: App.js [hunk 4]
```diff
@@ -1,26 +1,151 @@
...
+      .then(json => {
+        const randomIndex = Math.floor(Math.random() * Math.floor(json.results.length))
+        setJoke(json.results[randomIndex].joke)
+      })
+      .catch(() => {
+        alert("No such joke")
+      })
+      .finally(() => {
+        setSearch("")
+        setSearch()
```
Comment (EmOlsz) [inline App.js:104]: I think you don't need one of those lines

**Review by EmOlsz — CHANGES_REQUESTED** [review]

File: App.js (hunk 1 above)
Comment (ankimelin) [reply App.js:87]: makes sense, updated! 

also updated search and setSearch to searchWord and setSearchWord

File: App.js (hunk 2 above)
Comment (ankimelin) [reply App.js:114]: fixed!

File: App.js (hunk 3 above)
Comment (ankimelin) [reply App.js:108]: thanks!

File: App.js (hunk 4 above)
Comment (ankimelin) [reply App.js:104]: Actually I do, setSearch("") (now setSearchWord("")) will empty the textinput field, but the state will hold and empty string which is totally fine to query jokes on from the api - I don't want that. So what setSearch() (not setSearchWord()) does is to empty the state, with this one alone the textinput field is not emptied - only the state is emptied.

**Comment by ankimelin** [comment]
I have updated/answered on the comments requesting changes, please let me know if the project is goot to go!

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**